Framework/DataHandling/src/LoadGSS.cpp

```diff
@@ -187,33 +194,51 @@ API::MatrixWorkspace_sptr LoadGSS::loadGSASFile(const std::string &filename,
         }
       } else if (key1 == "Primary") {
         // Primary flight path ...
-        std::string s1, s2;
-        inputLine >> s1 >> s2;
-        primaryflightpath = atof(s2.c_str()); // convertToDouble(s2);
-        g_log.information() << "L1 = " << primaryflightpath << '\n';
+        boost::smatch result;
+        if (boost::regex_search(inputLine.str(), result, L1_REG_EXP) &&
+            result.size() == 2) {
+          primaryflightpath = atof(std::string(result[1]).c_str());
```

Comment:
Could [std::stof](http://en.cppreference.com/w/cpp/string/basic_string/stof) be used here, ie.  `primaryflightpath = stof(std::string(result[1]));`?